server/manager/src/main/java/org/apache/accumulo/master/TabletGroupWatcher.java

```diff
@@ -230,117 +232,25 @@ public void run() {
           sendChopRequest(mergeStats.getMergeInfo(), state, tls);
           sendSplitRequest(mergeStats.getMergeInfo(), state, tls);
 
-          // Always follow through with assignments
-          if (state == TabletState.ASSIGNED) {
-            goal = TabletGoalState.HOSTED;
-          }
+          // Always follow through with assignments will give AssignedTabletHostedGoal
 
           // if we are shutting down all the tabletservers, we have to do it in order
-          if (goal == TabletGoalState.SUSPENDED && state == TabletState.HOSTED) {
+          // TODO figure out what this is optimization is doing
+          if (goal instanceof HostedTabletNotHostedGoal) {
             if (master.serversToShutdown.equals(currentTServers.keySet())) {
               if (dependentWatcher != null && dependentWatcher.assignedOrHosted() > 0) {
-                goal = TabletGoalState.HOSTED;
+                goal = new HostedTabletHostedGoal(master, tls);
```

Comment:
It looks like this optimization is only when shutting down entire cluster but I am not sure what it is doing or why.  We remove the goal if the tablet is trying to suspend when shutting down.  This is also the only spot that ```dependentWatcher``` is used.  I feel like this could be done better.

Reply:
Master.getGoalState() which is called before this, returns a ```goal = TabletGoalState.SUSPENDED``` if the tablet is on the serversToShutdown.  Which this check is most likely overriding... if its in the list for this tablet's dependentWatcher?  Ugh this code is nearly impossible to follow.